### PR TITLE
Added diffuse fading animation effect

### PR DESCRIPTION
**Pull request by sarbes**
## Description
Often enough, skinners use featureless textures to color in surfaces. This patch allows to animate diffuse colors to do the same effect a lot cheaper.

How to fade betweeen red and green:
`<animation effect="fadediffuse" start="FFFF0000" end="FF00FF00" time="200" tween="sine">WindowOpen</animation>
<animation effect="fadediffuse" start="FF00FF00" end="FFFF0000" time="200" tween="sine">WindowClose</animation>`

Colors are in AARRGGBB notation.

## Motivation and context
Using textures instead of a animating a color like this feels like a hack and wastes a lot of resources. At best, around 25% of rendering resources could be saved (depending on the skin).

## How has this been tested?
Not much, I'm not a skinner. I've submitted this PR to let it build, so that it can be tested.

## What is the effect on users?
None, as it is a new element. It alters alpha fading a bit, but it should be fine.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by sarbes** [comment]
I've improved the concept. In my eyes, it is ready. Here is the animation in action:
diffusefade

I've ignored the style format in one place where it wanted me to reorder a matrix arrray into separate lines. This would reduce the readability IMO.

File: xbmc/utils/ColorUtils.cpp [hunk 1]
```diff
@@ -131,3 +131,13 @@ bool UTILS::COLOR::comparePairColorInfo(const std::pair<std::string, ColorInfo>&
...
+
+ColorFloats UTILS::COLOR::ConvertToFloats(const Color argb)
+{
+  ColorFloats c;
+  c.alpha = static_cast<float>((argb >> 24) & 0xFF) * (1.0f / 255.0f);
+  c.red = static_cast<float>((argb >> 16) & 0xFF) * (1.0f / 255.0f);
+  c.green = static_cast<float>((argb >> 8) & 0xFF) * (1.0f / 255.0f);
+  c.blue = static_cast<float>(argb & 0xFF) * (1.0f / 255.0f);
+  return c;
+}
```
Comment (garbear) [inline xbmc/utils/ColorUtils.cpp:143]: Missing newline

**Review by garbear — APPROVED** [review]
No surprises in the code. Usually I apply a clang-format commit first, then I can add my changes on top of it and get no issues. Cool feature, and great performance boost!

File: xbmc/utils/ColorUtils.cpp (hunk 1 above)
Comment (sarbes) [reply xbmc/utils/ColorUtils.cpp]: Done.

**Comment by sarbes** [comment]
@garbear  one last green light?

**Review by garbear — APPROVED** [review]
:+1:

**Comment by ksooo** [comment]
@sarbes next time, before merging, please fix the code formatting changes requested by jenkins4kodi.

**Pull request merged**